### PR TITLE
User swagger and leftover endpoints of wishList(delete by Id) and enrolledCourses(delete and update by Id)

### PR DESCRIPTION
**Pull request by Devesh21700Kumar**
1). Completed API response on create and update user to
2). Completed the API response body part on user controller

npm run start:dev and npm run test used for testing correctness

swagger docs checked in http://localhost:5000/api/  for ensuring correct fill

@Abhishek-kumar09  please suggest changes

### CODE REVIEW

**Review by Abhishek-kumar09 — COMMENTED** [review]
https://github.com/codeforcauseorg/edu-server/pull/85/files#diff-d3d9391580c34e207b691af4f5d16453866cdb4a256b38f2109bb814114f06cfL17-R48
Remove it from here and put those in a new folder

**Comment by Devesh21700Kumar** [comment]
> https://github.com/codeforcauseorg/edu-server/pull/85/files#diff-d3d9391580c34e207b691af4f5d16453866cdb4a256b38f2109bb814114f06cfL17-R48
> Remove it from here and put those in a new folder

Which one do i remove? I had just run npm run lint:fix and that changed occured (the link refers to that change)

**Comment by Devesh21700Kumar** [comment]
> https://github.com/codeforcauseorg/edu-server/pull/85/files#diff-d3d9391580c34e207b691af4f5d16453866cdb4a256b38f2109bb814114f06cfL17-R48
> Remove it from here and put those in a new folder

I did not understand this one. What do i move?

**Comment by Abhishek-kumar09** [comment]
https://github.com/codeforcauseorg/edu-server/blob/85de7b0189ff66561270127fc9a32a6be0ef4873/src/modules/user/user.controller.ts#L17-L47

Oh I was referring to this, Move it to new Folder named like Response Schema.

**Comment by Devesh21700Kumar** [comment]
> https://github.com/codeforcauseorg/edu-server/blob/85de7b0189ff66561270127fc9a32a6be0ef4873/src/modules/user/user.controller.ts#L17-L47
> 
> Oh I was referring to this, Move it to new Folder named like Response Schema.

ok Sure.

**Comment by Devesh21700Kumar** [comment]
> https://github.com/codeforcauseorg/edu-server/blob/85de7b0189ff66561270127fc9a32a6be0ef4873/src/modules/user/user.controller.ts#L17-L47
> 
> Oh I was referring to this, Move it to new Folder named like Response Schema.

Done

**Comment by Devesh21700Kumar** [comment]
@Abhishek-kumar09 I think that this can be merged, since the changes were made

**Review by Abhishek-kumar09 — COMMENTED** [review]
There are linting errors, Please fix those.

**Comment by Devesh21700Kumar** [comment]
@Abhishek-kumar09 the errors can't be fixed as npm run lint:fix did no change and no red bar is showing

File: src/firebase/firebase.module.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { Module } from '@nestjs/common';
-import { FirebaseAdminModule } from '@tfarras/nestjs-firebase-admin';
+import { FirebaseAdminCoreModule } from '@tfarras/nestjs-firebase-admin';
```
Comment (Abhishek-kumar09) [inline src/firebase/firebase.module.ts:2]: Why this change,? It was deprecated right.

Comment (Devesh21700Kumar) [reply src/firebase/firebase.module.ts:2]: I resolved some conflicts that had arisen after ur PR.. and did npm install and it started to show correctly

**Comment by Devesh21700Kumar** [comment]
@Abhishek-kumar09 

1). Completed all API endpoints of user, courses, enrolled and wishlist courses Namely
A). Get Enrolled By Id
B). Update Enrolled by Id
C). Delete Enrolled by Id
D). Deleted wishlist by Id

2). Completed the swagger documentation with better naming

Also in the user service file , the JEST tests are going on side by side which will be updated in  the next PR

Please see this one and suggest any changes and merge if fine

File: src/firebase/firebase.module.ts (hunk 1 above)
Comment (Abhishek-kumar09) [reply src/firebase/firebase.module.ts:2]: Revert this change. 
Delete your node_modules.
then run npm install
`@tferras/firebase ` is using version version 2, where FireAdminCoreModule is not available.

Comment (Devesh21700Kumar) [reply src/firebase/firebase.module.ts]: Done

File: src/modules/user/dto/create-enrolled.dto.ts [hunk 2]
```diff
@@ -8,16 +9,45 @@ interface video {
...
+    default: '',
+  })
   studentId: mongoose.Schema.Types.ObjectId;
 
+  @ApiProperty({
+    type: [Boolean],
+    description: 'The description of the student',
+    default: '',
+  })
   videosWatched: boolean[];
```
Comment (Abhishek-kumar09) [inline src/modules/user/dto/create-enrolled.dto.ts:24]: Description is not matching the property.

File: src/modules/user/dto/create-enrolled.dto.ts [hunk 3]
```diff
@@ -8,16 +9,45 @@ interface video {
...
 
+  @ApiProperty({
+    type: [Boolean],
+    description: 'The assignments doen by the student',
+    default: '',
+  })
   assignmentsDone: boolean[];
```
Comment (Abhishek-kumar09) [inline src/modules/user/dto/create-enrolled.dto.ts:31]: typo in `done`

File: src/modules/user/dto/create-enrolled.dto.ts [hunk 4]
```diff
@@ -8,16 +9,45 @@ interface video {
...
 
+  @ApiProperty({
+    description: 'The videos',
+    default: '',
+  })
   currentVideo: video[];
```
Comment (Abhishek-kumar09) [inline src/modules/user/dto/create-enrolled.dto.ts:37]: `The current video where student left`

File: src/modules/user/dto/create-user.dto.ts [hunk 5]
```diff
@@ -1,23 +1,80 @@
...
+    type: String,
+    description: 'The last name of the student',
+    default: '',
+  })
   readonly last_name: string;
 
   @IsNotEmpty()
+  @ApiProperty({
+    type: String,
+    description: 'The email of the student',
```
Comment (Abhishek-kumar09) [inline src/modules/user/dto/create-user.dto.ts:24]: `email ID`

File: src/modules/user/dto/create-user.dto.ts [hunk 6]
```diff
@@ -1,23 +1,80 @@
...
+  @ApiProperty({
+    type: String,
+    description: 'The address of the student',
+    default: '',
+  })
   readonly address: string;
 
+  @ApiProperty({
+    type: String,
+    description: 'The description of the student',
```
Comment (Abhishek-kumar09) [inline src/modules/user/dto/create-user.dto.ts:46]: `'The short description about the student'`

File: src/modules/user/dto/create-user.dto.ts [hunk 7]
```diff
@@ -1,23 +1,80 @@
...
 
+  @ApiProperty({
+    type: Number,
+    description: 'The score of the student',
+  })
   readonly score: number;
+
+  @ApiProperty({
+    type: Boolean,
+    description: 'The field to show of whether the students is Admin or not',
```
Comment (Abhishek-kumar09) [inline src/modules/user/dto/create-user.dto.ts:59]: `Whether the student is admin`
Also add default property which is false

File: src/modules/user/dto/update-enrolled.dto.ts [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+interface video {
+  num: number;
+  timestamp?: Date;
+}
+
+export class UpdateEnrolledDTO {
+  @ApiProperty({
+    type: [Boolean],
+    description: 'The description of the student',
+    default: '',
```
Comment (Abhishek-kumar09) [inline src/modules/user/dto/update-enrolled.dto.ts:12]: Why have you added default = `''` everywhere?

File: src/modules/user/dto/create-enrolled.dto.ts (hunk 2 above)
Comment (Devesh21700Kumar) [reply src/modules/user/dto/create-enrolled.dto.ts:24]: Fixed

File: src/modules/user/dto/create-enrolled.dto.ts (hunk 3 above)
Comment (Devesh21700Kumar) [reply src/modules/user/dto/create-enrolled.dto.ts:31]: Ok

File: src/modules/user/dto/create-enrolled.dto.ts (hunk 4 above)
Comment (Devesh21700Kumar) [reply src/modules/user/dto/create-enrolled.dto.ts:37]: Ok

File: src/modules/user/dto/create-user.dto.ts (hunk 5 above)
Comment (Devesh21700Kumar) [reply src/modules/user/dto/create-user.dto.ts:24]: Fixed

File: src/modules/user/dto/create-user.dto.ts (hunk 6 above)
Comment (Devesh21700Kumar) [reply src/modules/user/dto/create-user.dto.ts:46]: Done

File: src/modules/user/dto/create-user.dto.ts (hunk 7 above)
Comment (Devesh21700Kumar) [reply src/modules/user/dto/create-user.dto.ts:59]: Fixed

File: src/modules/user/dto/update-enrolled.dto.ts (hunk 8 above)
Comment (Devesh21700Kumar) [reply src/modules/user/dto/update-enrolled.dto.ts:12]: Oh, I will fill something here

Comment (Abhishek-kumar09) [reply src/modules/user/dto/update-enrolled.dto.ts]: @Devesh21700Kumar  What does default depicts?

**Comment by Abhishek-kumar09** [comment]
@Devesh21700Kumar What does default depicts?

File: src/modules/user/dto/update-enrolled.dto.ts (hunk 8 above)
Comment (Devesh21700Kumar) [reply src/modules/user/dto/update-enrolled.dto.ts]: i've added the defauts now.
i had not completed.
it just depicts an example value of the property nothing else

**Review by Abhishek-kumar09 — CHANGES_REQUESTED** [review]

File: src/modules/user/dto/create-user.dto.ts [hunk 9]
```diff
@@ -1,23 +1,83 @@
...
 import { IsNotEmpty } from 'class-validator';
+import * as mongoose from 'mongoose';
 
 export class CreateUserDTO {
   @IsNotEmpty()
+  @ApiProperty({
+    type: String,
+    description: 'The first name of the student',
+    default: 'Tony',
+  })
```
[the remark this answers is not in the document]
Comment (Devesh21700Kumar) [reply src/modules/user/dto/create-user.dto.ts:11]: Ok

File: src/modules/user/user.controller.ts [hunk 10]
```diff
@@ -64,29 +34,62 @@ export class UserController {
...
     @Param('userId') userId: mongoose.Schema.Types.ObjectId,
   ) {
     return await this.userService.getEnrolledCourses(userId);
   }
 
+  // retreiving enrolled course by id of the course and of a user
+  @Get('/:userId/enrolledCourses/:courseId')
+  @ApiCreatedResponse({ type: CreateEnrolledDTO })
+  async getEnrolledCoursesById(
+    @Param('userId') userId: mongoose.Schema.Types.ObjectId,
```
[the remark this answers is not in the document]
Comment (Devesh21700Kumar) [reply src/modules/user/user.controller.ts]: Ok

File: .eslintrc.js [hunk 11]
```diff
@@ -3,6 +3,7 @@ module.exports = {
   parserOptions: {
     project: 'tsconfig.json',
     sourceType: 'module',
+    extraFileExtensions: ['.json'],
```
Comment (Abhishek-kumar09) [inline .eslintrc.js:6]: revert this. This error was arising due to json file `example.service.json` but it is fixed in the previous commit.

File: src/modules/course/schema/schedule.schema.ts [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+import { Document } from 'mongoose';
+
+export type ScheduleDocument = Schedule & Document;
+
+@Schema({ timestamps: true })
+export class Schedule {
+  @Prop({ required: true })
+  chapterName: string;
+
+  @Prop({})
```
Comment (Abhishek-kumar09) [inline src/modules/course/schema/schedule.schema.ts:11]: make it required

File: src/modules/user/dto/create-enrolled.dto.ts [hunk 13]
```diff
@@ -10,7 +9,6 @@ interface video {
 export class CreateEnrolledDTO {
   @IsNotEmpty()
   @IsMongoId()
-  @ApiProperty({ type: Schema.Types.ObjectId })
```
Comment (Abhishek-kumar09) [inline src/modules/user/dto/create-enrolled.dto.ts:13]: revert all the changes done in this file.

**Review by Abhishek-kumar09 — APPROVED** [review]
